libs/core/src/lib/icon-tab-bar/directives/dnd/icon-bar-dnd-item.directive.ts

```diff
@@ -0,0 +1,166 @@
+
+    /**
+     * @description Classes for hovered state
+     */
+    @Input()
+    dndHoveredClass: string[] = ['fd-icon-tab-bar__item--dnd-hovered'];
+
+    /**
+     * @description Classes for separator, when draggable element between two tabs
+     */
+    @Input()
+    dndSeparatorClass: string[] = ['fd-icon-tab-bar__item--dnd-separator', 'fd-icon-tab-bar__item--dnd-separator-vertical'];
+
+    /**
+     * @description Event thrown when the element is moved by 1px
+     */
+    readonly moved = new EventEmitter<Point>();
+
+    /**
+     * @description Event thrown when the element is released
+     */
+    readonly released = new EventEmitter<void>();
+
+    /**
+     * @description Event thrown when the element is started to be dragged
+     */
+    readonly started = new EventEmitter<void>();
+
+    /** @hidden */
+    isVertical = false;
+
+    /**
+     * @hidden
+     * @description Drag reference, object created from DND CDK Service
+     */
+    dragRef: DragRef;
+
+    /** @hidden */
+    private readonly _onDestroy$ = new Subject<void>();
+
+    /** @hidden */
+    constructor(
+        public elementRef: ElementRef,
+        protected _dragDrop: DragDrop,
+        private _dndListDir: IconBarDndListDirective,
+        private _dndContainerDir: IconBarDndContainerDirective
+    ) {
+    }
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        this.isVertical = this._dndListDir?.dndOrientation === 'vertical';
+        this._setCDKDrag();
+    }
+
+    /** @hidden */
+    ngOnDestroy(): void {
+        this._dndListDir.removeDragItem(this);
+        this._dndContainerDir.removeDragItem(this);
+        this._onDestroy$.next();
+        this._onDestroy$.complete();
+    }
+
+    /** @hidden */
+    onCdkDragStart(): void {
+        /** Adds class */
+        this.started.emit();
+    }
+
+    /** @hidden */
+    onCdkMove(position: Point): void {
+        this.moved.emit(position);
+    }
+
+    /** @hidden */
+    onCdkDragReleased(): void {
+        this.released.emit();
+    }
+
+    /** @hidden */
+    getElementCoordinates(): ElementChord {
+        /** Takes distance from the beginning of window page */
+        const rect = <DOMRect>this.elementRef.nativeElement.getBoundingClientRect();
+
+        /** Vertically distance is counted by distance from top of the side + half of the element height */
+        return {
+            x: rect.left,
+            y: rect.top,
+            width: rect.width,
+            height: rect.height
+        };
+    }
+
+
+    /**
+     * @description Add/delete separator classes
+     */
+    toggleSeparatorStyles(force: boolean = false): void {
+        force
+            ? this.elementRef.nativeElement.classList.add(...this.dndSeparatorClass)
+            : this.elementRef.nativeElement.classList.remove(...this.dndSeparatorClass);
+    }
+
+    /**
+     * @description Add/delete hovered state classes
+     */
+    toggleHoveredStyles(force: boolean = false): void {
```

Comment:
Here and in other directives I see that many methods named like public (i.e. exposed from the library, without `_`), is it expected?

Especially about drag&drop methods, should the user have ability to use these methods?